apps/meteor/app/models/server/models/LivechatRooms.js

```diff
@@ -288,13 +288,14 @@ export class LivechatRooms extends Base {
 		return this.findOne(query, options);
 	}
 
-	findOneOpenByVisitorTokenAndDepartmentId(visitorToken, departmentId, options) {
+	findOneOpenByVisitorTokenAndDepartmentIdAndSource(visitorToken, departmentId, source, options) {
 		const query = {
 			't': 'l',
 			'open': true,
 			'v.token': visitorToken,
 			departmentId,
 		};
+		if (source) query['source.type'] = source;
```

Comment:
We don't use short ifs 🙈 (I like them, but it's on the styleguide haha)
```suggestion
		if (source) {
			query['source.type'] = source;
		}
```

Reply:
fixed!